File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/CreatedTenantImpulse.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * Impulse which indicates the result of a create Tenant impulse
 */
class CreatedTenantImpulse: Type() {

    fun id() { 
        add {"id" } 
    }

    /**
     * the Tenant instance
     */
    fun data(config: Tenant.() -> Unit) = 
        add { "data{ " + Tenant().apply(config).render(it) + "}"}

}

File: gaia-sdk-javascript/src/graphql/response/type/CreatedTenantImpulse.ts

import {Tenant} from "./Tenant";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result of a create Tenant impulse
*/
export interface CreatedTenantImpulse {
    id?:Uuid, 
    /**
    * the Tenant instance
    */
    data?:Tenant
}